packages/components/src/DetailTransactionButton.tsx

```diff
@@ -95,9 +102,41 @@ export class DetailTransactionButton extends React.Component<
   }
 
   public async componentDidMount(): Promise<void> {
+    this.createEthereumSubscription(this.props.civil);
     await this.divinePrice(this.props.estimateFunctions);
   }
 
+  public async componentWillUnmount(): Promise<void> {
+    this.unbscribeEthereum();
+  }
+
+  public createEthereumSubscription(civil?: Civil): void {
```

Comment:
this stuff is good but it should all happen outside of the component and be stored in redux, like how we store network/account in redux currently

Reply:
@nickreynolds There is no redux in the components package and it's used in multiple places, not only the dapp. This is the most backwards-compatible way I've found. If we get eg a standard of how we should write react components and refactor all our front-ends then sure
cc @dankins we really need one

Reply:
well you would just pass this data in to the component as a prop and the DApp would get it from its redux